lib/rules/jsx-no-update-forcing-props.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @fileoverview Prevent usage of props that would cause the component to render every time its parent renders
+ * @author Wojciech Maj
+ */
+
+'use strict';
+
+const propName = require('jsx-ast-utils/propName');
+const docsUrl = require('../util/docsUrl');
+const jsxUtil = require('../util/jsx');
+
+const typeToName = {
+  object: 'Objects',
+  array: 'Arrays',
+  func: 'Functions',
+  arrowFunc: 'Functions',
+  new: 'Constructor expressions',
+  call: 'Function calls'
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent usage of props that would cause the component to render every time its parent renders',
+      category: 'Best Practices',
+      recommended: false,
+      url: docsUrl('jsx-no-update-forcing-props')
+    },
+    schema: [{
+      type: 'object',
+      properties: {
+        objectExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        arrayExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        functionExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        arrowFunctionExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        newExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        callExpressions: {
+          default: false,
+          type: 'boolean'
+        },
+        ignoreRefs: {
+          default: true,
+          type: 'boolean'
+        }
```

Comment:
maybe rather than 7 booleans, 6 of which are named "allow" and default to false, and the 7th named "ignore" and defaults to true - we could use an object like:
```
{
  functions: 'allow',
  arrowFunctions: 'allow',
  objects: 'allow',
  arrays: 'allow',
  newExpressions: 'allow',
  callExpressions: 'allow',
  refs: 'ignore',
}
```
?

Reply:
The problem is that ignoreRefs isn't like other flags. If true (default), it will ignore checking ref prop for all of the above value types.

Reply:
Ahhh ok, it’s like a parallel setting. Comment still applies to the 6 booleans tho; seems like an object might be easier to manage?

Reply:
I thought it's already an object with a bunch of properties? I'm a noob when it comes to building ESLint rules though.

Reply:
You're right that the config object has a bunch of properties - i'm saying that rather than add them at the top level, it might be better to nest them under a single property, so that *future* options to the rule can go alongside that single property.